tests/rbd/test_rbd_trash_purge_schedule.py

```diff
@@ -0,0 +1,152 @@
+import random
+import time
+
+from ceph.parallel import parallel
+from tests.rbd.rbd_utils import Rbd, initial_rbd_config
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def create_an_image_run_fio(rbd, pool, client, image_name):
+    if rbd.create_image(pool, image_name, "10G"):
+        log.error(f"Image creation failed for {image_name}")
+        return 1
+    run_fio(image_name=image_name, pool_name=pool, client_node=client, runtime=10)
+    return 0
+
+
+def create_images_run_fio_in_parallel(rbd, pool, client, num_images):
+    log.info(f"Creating {num_images} images in pool {pool} and initiating io")
+
+    with parallel() as p:
+        for i in range(0, num_images):
+            p.spawn(create_an_image_run_fio, rbd, pool, client, f"image_{i}")
+        return 0
+
+
+def move_all_images_to_trash(rbd, pool, num_images):
+    log.info("Moving all images in pool to trash")
+    for i in range(0, num_images):
+        rbd.move_image_trash(pool, f"image_{i}")
+
+
+def prepare_images_and_move_to_trash(rbd, pool, client, num_images):
+    create_images_run_fio_in_parallel(rbd, pool, client, num_images)
+    move_all_images_to_trash(rbd, pool, num_images)
+
+
+def verify_schedule_add(
+    rbd: Rbd,
+    verification_iterations: int,
+    schedule_interval: int,
+    pool: str,
+    client,
+    num_images: int,
+) -> int:
+    iterations_remaining = verification_iterations
+    log.debug(f"Verifying purge on pool {pool}")
+    while iterations_remaining > 0:
+        log.debug(f"Sleeping for {schedule_interval} mins with  buffer")
+        time.sleep(schedule_interval * 60 + 5)
+        trash_list = rbd.trash_list(pool)
+        log.debug(f"Trash list: {trash_list}")
+        if len(trash_list) > 3:
+            log.error("Images found in trash unexpectedly")
+            return 1
+        prepare_images_and_move_to_trash(rbd, pool, client, num_images)
+        iterations_remaining -= 1
+
+    log.info(
+        f"Verified trash purge schedule of {schedule_interval} minutes for {verification_iterations} iterations"
+    )
+    return 0
+
+
+def rbd_trash_purge_schedule(rbd, pool_type, **kw):
+    """
+    Orchestrate rbd trash purge schedule test
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config and/or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    num_images = random.randint(2, 5)
+    schedule_interval = random.randint(1, 5)
+    verification_iterations = random.randint(2, 5)
+
+    log.info(
+        f"Trash purge schedule will be tested for an interval of \
```

Comment:
please modify \ is not required ....its creating spacing in log.

Reply:
Rajendra, we need to split the line to avoid tox failures and I've tried to rectified the spacing concern in upcoming rebased push.